src/targets/gpu/mlir.cpp

```diff
@@ -636,7 +628,11 @@ struct mlir_program
         return op;
     }
 
-    void find_target() { target_arch = get_device_name(); }
+    void get_gpu_properties()
+    {
+        target_arch = get_device_name();
+        num_cu      = get_cu_count();
```

Comment:
This should use `ctx.get_current_device().get_cu_count()`. Ideally we should do the same for device name.